src/GenerateB.c

```diff
@@ -181,10 +177,89 @@ void InitBeam(void)
 					default: LogError(ONE_POS,"Incompatibility error in GenerateB");
 				}
 				beam_descr=dyn_sprintf("Gaussian beam (%s)\n"
-				                       "\tWidth="GFORMDEF" (confinement factor s="GFORMDEF")\n"
-				                       "\tCenter position: "GFORMDEF3V,tmp_str,w0,s,COMP3V(beam_center_0));
+				                       "\tWidth="GFORMDEF" (confinement factor s="GFORMDEF")\n",tmp_str,w0,s);
+			}
+			return;
+#ifndef NO_FORTRAN
+		case B_BES_CS:
+		case B_BES_CSp:
+		case B_BES_M:
+		case B_BES_LE:
+		case B_BES_LM:
+		case B_BES_TEL:
+		case B_BES_TML:
+			if (surface) PrintError("Currently, Bessel incident beam is not supported for '-surf'");
+			// initialize parameters
+			n0=round(beam_pars[0]);
+			alpha0=beam_pars[1];
+			K =fabs(WaveNum);
+			Kt=fabs(WaveNum)*sin(alpha0);
+			Kz=fabs(WaveNum)*cos(alpha0);
+			switch (beamtype) { // definition of elements of matrix M ((Mex,Mey),(Mmx,Mmy))
+				case B_BES_CS:
+					M[0]=0.5;	M[1]=0;
+					M[2]=0;		M[3]=0.5;
+					break;
+				case B_BES_CSp:
+					M[0]=0.5;	M[1]=0;
+					M[2]=0;		M[3]=-0.5;
+					break;
+				case B_BES_M:
+					M[0]=beam_pars[2]+I*beam_pars[6];
+					M[1]=beam_pars[3]+I*beam_pars[7];
+					M[2]=beam_pars[4]+I*beam_pars[8];
+					M[3]=beam_pars[5]+I*beam_pars[9];
```

Comment:
Imaginary parts are not necessarily defined in the command line. The number of beam parameters should be checked before this code to avoid undefined behavior (they are not necessarily zero).